### ISSUE TITLE
Block all/unblock all from command line

### ISSUE THREAD

**Issue by jappeace** [cgag/hostblock#8]
```
$ hostblock -b
```

blocks all items in the list

```
$ hostblock -u
```

spawns the typeover sentence promt to unblock.

I think these are the most common use cases, you want to not visit certain sites while working/learning and you have a preconfigured list to do that, ie reddit, youtube etc. So being able to do that from the commandline without entering the menu (and selecting/deselecting everything) saves seconds.

**Comment by cgag** [comment]
Interesting idea, I kind of like it, but I also like being discriminate in what I unblock, even when I'm not working on anything in particular just because I think if I'm going to goof off I should go somewhere purposeful rather than polling reddit.  I'm not sure I want to give myself an easy way to avoid the guilt I currently have when I've unblocked most of my list, even if I don't think I'd use it.

Though I do agree the use case you describe is probably common even if it's not my style.  I'll think about it more. 

Thanks a bunch for the suggestion either way, I really appreciate feedback and ideas.

**Comment by theRealPadster** [comment]
It also might be a good idea to make block all an easy option (e.g. ublock -b) and optionally keep it a manual process to unblock. This would dissuade people (read: _me_) from just using the unblock all option when they still have work to do.

**Comment by jappeace** [comment]
@theRealPadster The implementation I wrote today ([pull request](https://github.com/cgag/hostblock/pull/9)) does blocking easy (ie no passphrase) and unblocking requires _always_ a passphrase. So you can skip the menu entirely unless you have to manage items or do fine grained blocking.
I'm assuming here that typing the passphrase is enough mental effort to stop you from slacking off. (and that just that is enough). 

If you think that unblocking becomes to easy by skipping the menu I suggest to lengthen the passprase instead.

**Comment by cgag** [comment]
Hmm I just commented saying I don't like -u on the PR, but making the passphrase longer might be an interesting compromise.  Do you mean just for when using -u or in general?  Any thoughts on that @theRealPadster?

**Comment by theRealPadster** [comment]
I personally have a more than 10 items on my list and it takes a little while to space+down arrow through them all. I feel like if I could just type out the passphrase it wouldn't be as much of a barrier. Making it longer could help, though personally I would rather have to go through them all. I don't speak for everyone though. 

If you make the passphrase longer, then it would just be for the -u option if implemented, no sense making it longer for the manual way. Would it be possible to have a config or compile option for enabling the unblock mode or not?

**Comment by jappeace** [comment]
@theRealPadster Yeah I'm going implement it that -u is just longer (2 words extra should be fine?), I also fix some other stuff that could be done better, then I'm making another pull request.

A compile option would also be an idea to just outright disable -u. (using this I think: https://doc.rust-lang.org/book/conditional-compilation.html). I guess I could add that to, disable or enable by default?

**Comment by theRealPadster** [comment]
@jappeace Sure, that would be a good compromise. A compile option would be good, then I could just disable it if I want.
